dm/syncer/error.go

```diff
@@ -61,6 +64,103 @@ func isDropColumnWithIndexError(err error) bool {
 			strings.Contains(mysqlErr.Message, "with tidb_enable_change_multi_schema is disable"))
 }
 
+// GetDDLStatusFromTiDB() retrieves the synchronizing status of DDL from TiDB
+// hence here db should be TiDB database.
+func GetDDLStatusFromTiDB(ctx context.Context, db *sql.DB, ddl string, createTime int64) (string, error) {
+	rowNum := 10
+	count := 0
+	for {
+		showJobs := fmt.Sprintf("ADMIN SHOW DDL JOBS %d", rowNum)
+		// every attempt try 10 history jobs
+		rows, err := db.QueryContext(ctx, showJobs)
+		if err != nil {
+			rows.Close()
+			return "", err
+		}
+
+		columns, err := rows.Columns()
```

Comment:
there is a function `export.GetSpecifiedColumnValuesAndClose` in `"github.com/pingcap/tidb/dumpling/export"` to implement read part of columns according to column names (not hardcoded column offset), maybe you can use it.